ert_gui/tools/export/exporter.py

```diff
@@ -15,65 +15,50 @@
-
-        if self.__export_keyword_model.isFieldKw(keyword):
-            self.exportField(keyword, file_name, values["iactive"], values["file_type_key"], values["report_step"], values["selected_case"])
-        elif self.__export_keyword_model.isGenKw(keyword):
-            self.exportGenKw(keyword, file_name, values["iactive"], values["file_type_key"], values["report_step"], values["selected_case"])
-        elif self.__export_keyword_model.isGenParamKw(keyword) or self.__export_keyword_model.isGenDataKw(keyword):
-            self.exportGenData(keyword, file_name, values["iactive"], values["file_type_key"], values["report_step"], values["selected_case"])
-        else:
-            sys.stderr.write('** WARNING: Cannot export unknown keyword type "%s".\n' % keyword)
-
-    def exportField(self, keyword, file_name, iactive, file_type_key, report_step, selected_case):
-        if file_type_key == "Eclipse GRDECL":
-            file_type = EnkfFieldFileFormatEnum.ECL_GRDECL_FILE
-        else:
-            file_type = EnkfFieldFileFormatEnum.RMS_ROFF_FILE
-
-        result = ExportModel().exportField(keyword, file_name, iactive, file_type, report_step, selected_case)
-        if not result:
-            QMessageBox.warning(self, "Warning", '''Something did not work!''', QMessageBox.Ok)
-
-    def exportGenData(self, keyword, file_name, iactive, file_type_key, report_step, selected_case):
-        ExportModel().exportGenData(keyword, file_name, iactive, file_type_key, report_step, selected_case)
-
-    def exportGenKw(self, keyword, file_name, iactive, file_type_key, report_step, selected_case):
-        ExportModel().exportGenKw(keyword, file_name, iactive, file_type_key, report_step, selected_case)
-
-    def createExportFileNameMask(self, keyword, current_case, report_step, path):
-        impl_type = None
-
-        if self.__export_keyword_model.isFieldKw(keyword):
-            impl_type = self.__export_keyword_model.getImplementationType(keyword)
-        elif self.__export_keyword_model.isGenDataKw(keyword):
-            impl_type = "Gen_Data"
-        elif self.__export_keyword_model.isGenKw(keyword):
-            impl_type = "Gen_Kw"
-        elif self.__export_keyword_model.isGenParamKw(keyword):
-            impl_type = "Gen_Param"
-
-        path = os.path.join(str(path), str(current_case), str(impl_type), str(keyword))
-
-        if self.__export_keyword_model.isGenDataKw(keyword):
-            path = path + "_" + str(report_step)
-
-        if not QDir(path).exists():
-            os.makedirs(path)
-
-        return path
+        self._export_job = "CSV_EXPORT2"
+        self._runpath_job = "EXPORT_RUNPATH"
+
+    def is_valid(self):
+        export_job = ERT.enkf_facade.get_workflow_job(self._export_job)
+        runpath_job = ERT.enkf_facade.get_workflow_job(self._runpath_job)
+
+        if export_job is None:
+            error = "Export not available due to {job_name} is not installed.".format(job_name=self._export_job)
```

Comment:
Do not swallow this error if we have a dependency on semeio.

Reply:
The functionality is depending on semeio, but we want to give an informed message if export is not available .